Controllers/StudentsController.cs

```diff
@@ -0,0 +1,33 @@
+using Microsoft.AspNetCore.Mvc;
+using StudentService.Data;
+using StudentService.Models;
+
+namespace StudentService.Controller
+{
+    [Route("api/{controller}")]
+    [ApiController]
+    public class StudentsController : ControllerBase
+    {
+        private readonly IStudentRepo _studentService;
+
+        public StudentsController(IStudentRepo studentService)
+        {
+            _studentService = studentService;
+        }
+
+        //POST api/students
+        [HttpPost]
+        public ActionResult <Student> CreateStudent(Student student){
+            if(student == null){
+                return BadRequest();
+            }
+            _studentService.CreateStudent(student);
```

Comment:
the implementation of the createStudent method in the service should determine how to save the student;
The saving of the student shouldn't be exposed to the service.
Do you understand the concept of low coupling and high cohesion in Software

Reply:
I do but I prolly need to study them more.

Reply:
I have loosen the coupling and maintained the inheritance for cohesion.